### DISCUSSION TITLE
Few seconds after starting things break

### DISCUSSION THREAD

**Discussion by LeonardoMor** [akinomyoga/ble.sh#462] [Q&A] [answered]
I use blesh on WSL2 with Arch Linux.

Whenever I turn on my computer and open an Arch CLI, things load and appear to be fine. Then after a few seconds, every key press results in this:

![image](https://github.com/akinomyoga/ble.sh/assets/66377812/37202c9f-df1a-48ae-867d-221d67c97736)

So there you can see I was in the middle of typing `exit`.

Then I have to close this Tab and open a new one. The second Tab doesn't do this and I can work normally. Not sure why this is happening. Help is appreciated.

**Comment by akinomyoga** [comment]
Exactly the same issue but for `/run/user` was reported in #426, and it turned out that WSL's `/run/user` is broken [1-7]. I added a workaround not to use `/run/user` in WSL.

- [1] https://github.com/microsoft/WSL/issues/9689
- [2] https://github.com/microsoft/WSL/issues/10473
- [3] https://github.com/microsoft/WSL/issues/10496
- [4] https://github.com/microsoft/WSL/issues/10498
- [5] https://github.com/microsoft/WSL/issues/10552
- [6] [WSLでsystemctl status --userがNo such file or directoryになる / Failed to connect to a Wayland serverとなってWalyandにつながらない - ぶていのログでぶログ](https://tech.buty4649.net/entry/2023/10/10/112349)
- [7] [Nvim always logs `server_start:164` error in WSL (SOLVED: permissions issue) · Issue #26058 · neovim/neovim](https://github.com/neovim/neovim/issues/26058)

Maybe `/tmp` is also broken in WSL.

**Reply by akinomyoga** [reply]
Hmm, according to https://github.com/microsoft/WSL/issues/8441#issuecomment-1139434972, the contents of `/tmp` seem to be cleared after a few seconds since the WSL system boots up. This is consistent with your observation. The problem doesn't seem to be fixed because the issue was just closed, and there is no mention after that. Maybe I need to avoid using even `/tmp` in WSL.

**Reply by akinomyoga** [reply]
https://github.com/microsoft/WSL/issues/8441#issuecomment-1921176856 says that a new issue for WSL should be created if the problem persists. I thought about creating an issue, but they require the version information of Windows, WSL, Linux kernel, Distro, etc. I don't have detailed information about your environment. @LeonardoMor Could you create an issue at WSL? I think you can use https://github.com/microsoft/WSL/issues/8441 as a template. You can also include mentions of https://github.com/microsoft/WSL/issues/8441, this discussion, etc.

**Reply by akinomyoga** [reply, marked as answer]
I added a workaround in fcbf1ed0e417433d0e56cf90cad111852115dbe2. You can update ble.sh by running `ble-update` and see if the problem is solved. (If you use v0.3 of `ble.sh`, please switch to 0.4.)

@LeonardoMor Have you created an issue in the upstream WSL repository? If not, maybe I can create one if you could provide the version information of Windows, WSL, Linux kernel, Distro in your environment.

**Reply by LeonardoMor** [reply]
Created issue [11739](https://github.com/microsoft/WSL/issues/11739)

**Reply by akinomyoga** [reply]
Thank you!
